File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOGpsStatusModel.kt
//
//  IDOGpsStatusModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

///
/// Get GPS Status event number

open class IDOGpsStatusModel(gpsRunStatus: Int, agpsIsValid: Int) : IDOBaseModel {

    /// GPS running status
    /// 0 - Not running
    /// 1 - Searching for satellites
    /// 2 - Tracking
    @SerializedName("gps_run_status")
    var gpsRunStatus: Int = gpsRunStatus

    /// Validity of AGPS (Assisted GPS) in hours
    /// Non-zero values indicate validity
    @SerializedName("agps_is_valid")
    var agpsIsValid: Int = agpsIsValid


    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}
    

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOGpsStatusModel.swift
//
//  IDOGpsStatusModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/12.
//

import Foundation
/// Get GPS Status event number
@objcMembers
public class IDOGpsStatusModel: NSObject, IDOBaseModel {
    /// GPS running status
    /// 0 - Not running
    /// 1 - Searching for satellites
    /// 2 - Tracking
    public var gpsRunStatus: Int
    /// Validity of AGPS (Assisted GPS) in hours
    /// Non-zero values indicate validity
    public var agpsIsValid: Int

    enum CodingKeys: String, CodingKey {
        case gpsRunStatus = "gps_run_status"
        case agpsIsValid = "agps_is_valid"
    }

    public init(gpsRunStatus: Int, agpsIsValid: Int) {
        self.gpsRunStatus = gpsRunStatus
        self.agpsIsValid = agpsIsValid
    }

    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}
